### PR TITLE
Update manga info when browsing if not in library

### LINKED ISSUE #957: [Bug] Suwayomi Database storing old metadata (bato source)
**PLEASE READ THIS**

I acknowledge that:

- I have updated to the latest version of the app.
- I have tried the troubleshooting guide described in `README.md`
- If this is a request for adding/changing an extension it should be brought up to your extension repo.
- If this is an issue with some extension not working properly, It does work inside Tachiyomi as intended.
- I have searched the existing issues and this is a new ticket **NOT** a duplicate or related to another open issue
- I will fill out the title and the information in this template

Note that the issue will be automatically closed if you do not fill out the title or requested information.

---

## Device information
- Suwayomi-Server version: v1.0.0-r1527-docker)
- Server Operating System: docker
- Server Desktop Environment: docker
- Server JVM version: docker
- Client Operating System: docker
- Client Web Browser: Chrome (newest version)

## Steps to reproduce
1. Open bato library
2. Some thumbnails don't show up when browsing the source

### Expected behavior
All thumbnails should be loaded while browsing library

### Actual behavior
Some thumbnails don't load - they error out and even reply doesn't work. To reload a picture you have to open the manga/manhwa with the thumbnail/cover problem and refresh it from inside - then it works correctly
It seems the server tries to connect to a CDN server that doesn't exist anymore:

log:

> 

> 18:08:39.439 [DefaultDispatcher-worker-9] ERROR suwayomi.tachidesk.server.JavalinSetup -- IOException while handling the request
> 
> java.io.IOException: xfs-s117.batcg.org: Name or service not known
> 	at suwayomi.tachidesk.manga.impl.Manga.fetchHttpSourceMangaThumbnail(Manga.kt:291)
> 	at suwayomi.tachidesk.manga.impl.Manga.fetchHttpSourceMangaThumbnail$default(Manga.kt:271)
> 	at suwayomi.tachidesk.manga.impl.Manga$fetchMangaThumbnail$2.invokeSuspend(Manga.kt:320)
> 	at suwayomi.tachidesk.manga.impl.Manga$fetchMangaThumbnail$2.invoke(Manga.kt)
> 	at suwayomi.tachidesk.manga.impl.Manga$fetchMangaThumbnail$2.invoke(Manga.kt)
> 	at suwayomi.tachidesk.manga.impl.util.storage.ImageResponse.getImageResponse(ImageResponse.kt:70)
> 	at suwayomi.tachidesk.manga.impl.Manga.fetchMangaThumbnail(Manga.kt:319)
> 	at suwayomi.tachidesk.manga.impl.Manga.getMangaThumbnail(Manga.kt:365)
> 	at suwayomi.tachidesk.manga.controller.MangaController$thumbnail$2$1.invokeSuspend(MangaController.kt:94)
> 	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
> 	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:108)
> 	at kotlinx.coroutines.internal.LimitedDispatcher$Worker.run(LimitedDispatcher.kt:115)
> 	at kotlinx.coroutines.scheduling.TaskImpl.run(Tasks.kt:103)
> 	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:584)
> 	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:793)
> 	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:697)
> 	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:684)
> Caused by: java.net.UnknownHostException: xfs-s117.batcg.org: Name or service not known
> 	at java.base/java.net.Inet4AddressImpl.lookupAllHostAddr(Native Method)
> 	at java.base/java.net.InetAddress$PlatformNameService.lookupAllHostAddr(Unknown Source)
> 	at java.base/java.net.InetAddress.getAddressesFromNameService(Unknown Source)
> 	at java.base/java.net.InetAddress$NameServiceAddresses.get(Unknown Source)
> 	at java.base/java.net.InetAddress.getAllByName0(Unknown Source)
> 	at java.base/java.net.InetAddress.getAllByName(Unknown Source)
> 	at java.base/java.net.InetAddress.getAllByName(Unknown Source)
> 	at okhttp3.Dns$Companion$DnsSystem.lookup(Dns.kt:49)
> 	at okhttp3.internal.connection.RouteSelector.resetNextInetSocketAddress(RouteSelector.kt:169)
> 	at okhttp3.internal.connection.RouteSelector.nextProxy(RouteSelector.kt:132)
> 	at okhttp3.internal.connection.RouteSelector.next(RouteSelector.kt:74)
> 	at okhttp3.internal.connection.RealRoutePlanner.planConnect(RealRoutePlanner.kt:147)
> 	at okhttp3.internal.connection.RealRoutePlanner.plan(RealRoutePlanner.kt:67)
> 	at okhttp3.internal.connection.FastFallbackExchangeFinder.launchTcpConnect(FastFallbackExchangeFinder.kt:118)
> 	at okhttp3.internal.connection.FastFallbackExchangeFinder.find(FastFallbackExchangeFinder.kt:62)
> 	at okhttp3.internal.connection.RealCall.initExchange$okhttp(RealCall.kt:267)
> 	at okhttp3.internal.connection.ConnectInterceptor.intercept(ConnectInterceptor.kt:32)
> 	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109)
> 	at okhttp3.internal.cache.CacheInterceptor.intercept(CacheInterceptor.kt:95)
> 	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109)
> 	at okhttp3.internal.http.BridgeInterceptor.intercept(BridgeInterceptor.kt:84)
> 	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109)
> 	at okhttp3.internal.http.RetryAndFollowUpInterceptor.intercept(RetryAndFollowUpInterceptor.kt:65)
> 	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109)
> 	at eu.kanade.tachiyomi.network.interceptor.CloudflareInterceptor.intercept(CloudflareInterceptor.kt:42)
> 	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109)
> 	at eu.kanade.tachiyomi.network.interceptor.UserAgentInterceptor.intercept(UserAgentInterceptor.kt:19)
> 	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109)
> 	at eu.kanade.tachiyomi.network.interceptor.UncaughtExceptionInterceptor.intercept(UncaughtExceptionInterceptor.kt:18)
> 	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109)
> 	at okhttp3.internal.connection.RealCall.getResponseWithInterceptorChain$okhttp(RealCall.kt:205)
> 	at okhttp3.internal.connection.RealCall$AsyncCall.run(RealCall.kt:537)
> 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
> 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
> 	at java.base/java.lang.Thread.run(Unknown Source)

EDIT: The problem is connected with Suwayomi storing old metadata in the database file.
For now the only fix is to backup your manga database and restore it on a fresh database

### PR DESCRIPTION
**Pull request by Syer10**
Closes #957

### CODE REVIEW

File: server/src/main/kotlin/suwayomi/tachidesk/manga/impl/MangaList.kt [hunk 1]
```diff
@@ -73,7 +76,42 @@ object MangaList {
...
+                    this[MangaTable.title] = sManga.title
+                    this[MangaTable.artist] = sManga.artist
+                    this[MangaTable.author] = sManga.author
+                    this[MangaTable.description] = sManga.description
+                    this[MangaTable.genre] = sManga.genre
+                    this[MangaTable.status] = sManga.status
+                    this[MangaTable.thumbnail_url] = sManga.thumbnail_url
+                    this[MangaTable.updateStrategy] = sManga.update_strategy.name
+                    if (!sManga.thumbnail_url.isNullOrEmpty() && manga[MangaTable.thumbnail_url] != sManga.thumbnail_url) {
+                        this[MangaTable.thumbnailUrlLastFetched] = Instant.now().epochSecond
```
Comment (schroda) [inline server/src/main/kotlin/suwayomi/tachidesk/manga/impl/MangaList.kt:98]: could `Manga.clearThumbnail` be called from here?

File: server/src/main/kotlin/suwayomi/tachidesk/manga/impl/MangaList.kt [hunk 2]
```diff
@@ -73,7 +76,42 @@ object MangaList {
...
+                    } else {
+                        this[MangaTable.thumbnailUrlLastFetched] = manga[MangaTable.thumbnailUrlLastFetched]
```
Comment (schroda) [inline server/src/main/kotlin/suwayomi/tachidesk/manga/impl/MangaList.kt:101]: is it required to set the current value?

Comment (Syer10) [reply server/src/main/kotlin/suwayomi/tachidesk/manga/impl/MangaList.kt:101]: Batch update statements require that you set the same columns in every batch, so we need to set a `MangaTable.thumbnailUrlLastFetched` even if its the same value

File: server/src/main/kotlin/suwayomi/tachidesk/manga/impl/MangaList.kt (hunk 1 above)
Comment (Syer10) [reply server/src/main/kotlin/suwayomi/tachidesk/manga/impl/MangaList.kt:98]: Probably

**Pull request merged**